control-plane/gateways/service.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package gateways
+
+import (
+	meshv2beta1 "github.com/hashicorp/consul-k8s/control-plane/api/mesh/v2beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const port = 443
+
+func (b *meshGatewayBuilder) Service() *corev1.Service {
+	var (
+		containerConfig *meshv2beta1.GatewayClassContainerConfig
+		PortModifier    = 0
+		serviceType     = corev1.ServiceType("")
+	)
+
+	if b.gcc != nil {
+		containerConfig = b.gcc.Spec.Deployment.Container
+		PortModifier = containerConfig.PortModifier
+		serviceType = *b.gcc.Spec.Service.Type
+	}
+
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      b.gateway.Name,
+			Namespace: b.gateway.Namespace,
+			Labels:    b.Labels(),
+			//TODO: Ask about annotations, shoule we set this?!
+			// Annotations: annotations,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: b.Labels(),
+			Type:     serviceType,
+			Ports: []corev1.ServicePort{
+				{
+					Name: b.gateway.Name,
```

Comment:
I think we should hardcode this to `wan` to match what we used for the container port that's targeted below